homeassistant/components/monarchmoney/sensor.py

```diff
@@ -0,0 +1,215 @@
+            "bicycle": "mdi:bicycle",
+            "other": "mdi:car",
+        },
+        "credit": {"credit_card": "mdi:credit-card"},
+        "depository": {
+            "cash_management": "mdi:cash",
+            "checking": "mdi:checkbook",
+            "savings": "mdi:piggy-bank-outline",
+            "money_market": "mdi:piggy-bank-outline",
+        },
+        "loan": {
+            "line_of_credit": "mdi:credit-card-plus",
+            "loan": "mdi:bank-outline",
+            "mortgage": "mdi:home-city-outline",
+        },
+    }
+
+    default_icons = {
+        "brokerage": "mdi:chart-line",
+        "credit": "mdi:credit-card",
+        "depository": "mdi:cash",
+        "loan": "mdi:bank-outline",
+        "vehicle": "mdi:car",
+    }
+    if account_subtype not in icon_mapping.get(account_type, {}):
+        LOGGER.info(
+            f"Unknown subtype '{account_subtype}' for account type '{account_type}'"
+        )
+        return default_icons.get(account_type, "mdi:cash")
+
+    account_type_icons = icon_mapping.get(account_type, {})
+    return account_type_icons.get(
+        account_subtype, default_icons.get(account_type, "mdi:help")
+    )
+
+
+@dataclass(frozen=True, kw_only=True)
+class MonarchMoneySensorEntityDescription(SensorEntityDescription):
+    """Describe a sensor entity."""
+
+    value_fn: Callable[[Any], StateType | datetime]
+    picture_fn: Callable[[Any], str] | None = None
+    icon_fn: Callable[[Any], str] | None = None
+
+
+MONARCH_MONEY_SENSORS: tuple[MonarchMoneySensorEntityDescription, ...] = (
+    MonarchMoneySensorEntityDescription(
+        key="currentBalance",
+        translation_key="balance",
+        state_class=SensorStateClass.TOTAL,
+        device_class=SensorDeviceClass.MONETARY,
+        value_fn=lambda account: account["currentBalance"],
+        picture_fn=lambda account: account["logoUrl"],
+        icon_fn=_type_to_icon,
+        native_unit_of_measurement=CURRENCY_DOLLAR,
+    ),
+    MonarchMoneySensorEntityDescription(
+        key="age",
+        translation_key="age",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda account: datetime.fromisoformat(account["updatedAt"]),
+    ),
+    MonarchMoneySensorEntityDescription(
+        key="created",
+        translation_key="created",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda account: datetime.fromisoformat(account["createdAt"]),
+    ),
+)
+
+MONARCH_CASHFLOW_SENSORS: tuple[MonarchMoneySensorEntityDescription, ...] = (
+    MonarchMoneySensorEntityDescription(
+        key="sum_income",
+        translation_key="sum_income",
+        value_fn=lambda summary: summary["sumIncome"],
+        state_class=SensorStateClass.TOTAL,
+        device_class=SensorDeviceClass.MONETARY,
+        native_unit_of_measurement=CURRENCY_DOLLAR,
+        icon="mdi:cash-plus",
```

Comment:
Use icon translations

Reply:
I don't think we can do that. The state is always going to be a numeric "thing". I tried to do this in SimpleFIN and it was a no-go.

Reply:
I can also do icons in a separate pr if this one is close.